File: scratch/test-api.js
async function test() {
  const loginUrl = 'https://techtopiagh-crm.onrender.com/api/v1/auth/login';
  const credentials = { email: 'admin@techtopia.com', password: 'Admin123!' };
  
  const loginRes = await fetch(loginUrl, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(credentials)
  });
  const data = await loginRes.json();
  const token = data.accessToken;

  // Probe without api/v1 prefix
  const paths = [
    '/finance/expenses/categories',
    '/finance/expense-categories',
    '/finance/categories',
    '/api/v1/finance/expenses/categories'
  ];

  for (const path of paths) {
    try {
      const res = await fetch(`https://techtopiagh-crm.onrender.com${path}`, {
        headers: {
          'Authorization': `Bearer ${token}`,
          'Tenant-Id': 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'
        }
      });
      console.log(`Path ${path} status: ${res.status}`);
      if (res.ok) {
        const body = await res.json();
        console.log(`Path ${path} categories (first 2):`, JSON.stringify(body.slice(0, 2), null, 2));
      }
    } catch (err) {
      console.log(`Path ${path} error:`, err.message);
    }
  }
}
test();

File: scratch/test-api.py
import requests
import json

def test():
    urls = [
        'https://techtopiagh-crm.onrender.com/openapi/v1.json',
        'https://techtopiagh-crm.onrender.com/swagger/v1/swagger.json'
    ]
    for url in urls:
        try:
            res = requests.get(url)
            print(f'URL: {url} status: {res.status_code}')
            if res.status_code == 200:
                data = res.json()
                print('Paths in OpenAPI:')
                paths = sorted(data.get('paths', {}).keys())
                for path in paths:
                    print(f'  {path}')
                # Also save the JSON for detailed review
                with open('scratch/openapi.json', 'w') as f:
                    json.dump(data, f, indent=2)
                print('Saved openapi.json to scratch/openapi.json')
                break
        except Exception as e:
            print('Error fetching:', url, e)

if __name__ == '__main__':
    test()
